tests/core/test_graph.py

```diff
@@ -1,54 +1,367 @@
+# pylint: disable=missing-function-docstring,missing-module-docstring
+# pylint: disable=invalid-name
+
 from __future__ import annotations
 
+import pytest
+import scipy.sparse as s
 import torch
+from torch_geometric.edge_index import EdgeIndex
+from torch_geometric.data import Data
+from torch_geometric.testing import get_random_edge_index
+
+from pathpyG import Graph, IndexMap
+
+
+def test_init():
+    edge_index = get_random_edge_index(100, 100, 1000)
+    data = Data(edge_index=edge_index, num_nodes=100)
+    g = Graph(data)
+    assert isinstance(g, Graph)
+    assert isinstance(g.data, Data)
+    assert isinstance(g.mapping, IndexMap)
+    assert isinstance(g.edge_to_index, dict)
+
+
+def test_init_with_edge_index():
+    edge_index = EdgeIndex(get_random_edge_index(100, 100, 1000))
+    data = Data(edge_index=edge_index, num_nodes=100)
+    g = Graph(data)
+    assert isinstance(g, Graph)
+    assert isinstance(g.data, Data)
+    assert isinstance(g.mapping, IndexMap)
+    assert isinstance(g.edge_to_index, dict)
+
+
+def test_init_with_mapping():
+    edge_index = get_random_edge_index(100, 100, 1000)
+    data = Data(edge_index=edge_index, num_nodes=100)
+    idxs = [str(i) for i in range(100)]
+    mapping = IndexMap(idxs)
+    g = Graph(data, mapping)
+    assert isinstance(g, Graph)
+    assert isinstance(g.data, Data)
+    assert isinstance(g.mapping, IndexMap)
+    assert isinstance(g.edge_to_index, dict)
+
+
+def test_init_with_list_mapping():
```

Comment:
Should we support initializing a `Graph` like this?

Reply:
Yes, this should be the default constructor for all core classes, i.e. initializing from a `pyG.data.Data` object, with additional static class methods to create instances from other data structures (which are mapped to `pyG.data.Data`,